sixtyfps_compiler/generator/cpp.rs

```diff
@@ -592,6 +598,34 @@ pub fn generate(doc: &Document, diag: &mut BuildDiagnostics) -> Option<impl std:
     file.includes.push("<cmath>".into()); // TODO: ideally only include this if needed (by floor/ceil/round)
     file.includes.push("<sixtyfps.h>".into());
 
+    file.declarations.extend(doc.root_component.embedded_file_resources.borrow().iter().map(
+        |(path, id)| {
+            let data =
+                std::fs::read(path).expect(&format!("unable to read file for embedding: {}", path));
+
+            let mut init = "{ ".to_string();
+
+            for (index, byte) in data.iter().enumerate() {
+                if index > 0 {
+                    init.push(',');
+                }
+                init.push_str(&format!("0x{:x}", byte));
```

Comment:
could the format! here be avoided to avoid many allocations?
I believe you can use `write!` instead.

Reply:
Ahh, `write!`. Thanks, that's what I was looking for but failed to find :)